python/ray/data/_internal/logical/operators/map_operator.py

```diff
@@ -177,7 +177,6 @@ def __init__(
         self._batch_format = batch_format
         self._zero_copy_batch = zero_copy_batch
 
-    @property
     def can_modify_num_rows(self) -> bool:
         return False
```

Comment:
for map_batches, it's safer to make it default to False. 
Because it's upper to the UDF to decide whether it can modify num rows. 
also, if we make this change, we can enable the limit pushdown rule now

Reply:
Not following your point -- it's already False.

For limit pushdown we'd still have to treat it like if it can modify since we simply have to assume the most pessimistic scenario.

Reply:
oops, I read that wrong. nvm